database.cc

```diff
@@ -1663,7 +1663,7 @@ database::make_keyspace_config(const keyspace_metadata& ksm) {
         }
         cfg.enable_disk_writes = !_cfg.enable_in_memory_data_store();
         cfg.enable_disk_reads = true; // we allways read from disk
-        cfg.enable_commitlog = ksm.durable_writes() && _cfg.enable_commitlog() && !_cfg.enable_in_memory_data_store();
+        cfg.enable_commitlog = _cfg.enable_commitlog() && !_cfg.enable_in_memory_data_store();
```

Comment:
Does it mean that all tables will have their commitlog enabled, but it just won't be used for some of them? What's the overhead of an empty, but enabled commitlog?

Reply:
Commitlog is shared across all tables on a shard (owned by `database`). But why not just go through all tables on schema change and simply null out the commitlog pointer iff durable writes is off, or add it if on?

Reply:
aye

Reply:
`table::set_schema` seems like a good candidate...
Will have to ensure that any call chain currently inserting data into the (obsolete schema version) of the table retains a pointer to the CL, but that should be fairly easy (since most of it is RAII).

Reply:
The commitlog pointer is still needed after durable_writes is turned off, because the table needs to clear its reference counters to the commitlog when the memtable is sealed (https://github.com/scylladb/scylla/blob/master/table.cc#L1528 and https://github.com/scylladb/scylla/blob/master/table.cc#L763).

Reply:
Yes, but you could just give that call access to the `database` object and use this commitlog pointer iff rp_set is non-empty. 
Or change rp_set to be fully RAII (i.e. keep track of CL). 
Or add a boolean var "actually put stuff into commitlog" and set it true/false in update_schema. 
Point being, you don't need to check "has durable writes" at every call. Can be done once per schema version.